PeakCoreData/Helpers/NSPredicate.swift

```diff
@@ -0,0 +1,268 @@
+            } else {
+                self.init(format: "%K != nil", argumentArray: [keyPath])
+            }
+        } else {
+            if let value = value {
+                self.init(format: "SELF != %@", argumentArray: [value])
+            } else {
+                self.init(format: "SELF != nil")
+            }
+        }
+    }
+    
+    /// Returns a predicate that performs a less than comparison.
+    ///
+    /// - Parameters:
+    ///   - keyPath: The keypath to the property. Pass in `nil` to use SELF.
+    ///   - value: The value to use in the comparison.
+    public convenience init(keyPath: KeyPath? = nil, lessThan value: Any) {
+        if let keyPath = keyPath {
+            self.init(format: "%K < %@", argumentArray: [keyPath, value])
+        } else {
+            self.init(format: "SELF < %@", argumentArray: [value])
+        }
+    }
+    
+    /// Returns a predicate that performs a less or equal to comparison.
+    ///
+    /// - Parameters:
+    ///   - keyPath: The keypath to the property. Pass in `nil` to use SELF.
+    ///   - value: The value to use in the comparison.
+    public convenience init(keyPath: KeyPath? = nil, lessThanOrEqualTo value: Any) {
+        if let keyPath = keyPath {
+            self.init(format: "%K <= %@", argumentArray: [keyPath, value])
+        } else {
+            self.init(format: "SELF <= %@", argumentArray: [value])
+        }
+    }
+    
+    /// Returns a predicate that performs a greater than comparison.
+    ///
+    /// - Parameters:
+    ///   - keyPath: The keypath to the property. Pass in `nil` to use SELF.
+    ///   - value: The value to use in the comparison.
+    public convenience init(keyPath: KeyPath? = nil, greaterThan value: Any) {
+        if let keyPath = keyPath {
+            self.init(format: "%K > %@", argumentArray: [keyPath, value])
+        } else {
+            self.init(format: "SELF > %@", argumentArray: [value])
+        }
+    }
+    
+    /// Returns a predicate that performs a greater than or equal to comparison.
+    ///
+    /// - Parameters:
+    ///   - keyPath: The keypath to the property. Pass in `nil` to use SELF.
+    ///   - value: The value to use in the comparison.
+    public convenience init(keyPath: KeyPath? = nil, greaterThanOrEqualTo value: Any) {
+        if let keyPath = keyPath {
+            self.init(format: "%K >= %@", argumentArray: [keyPath, value])
+        } else {
+            self.init(format: "SELF >= %@", argumentArray: [value])
+        }
+    }
+}
+
+// MARK: - Count
+
+extension NSPredicate {
+    
+    /// Returns a predicate that performs an equals comparison on a relationship count.
+    ///
+    /// - Parameters:
+    ///   - keyPath: The keypath of the entity relationship. Pass in `nil` to use SELF.
+    ///   - count: The count to use in the comparison.
+    public convenience init(keyPath: KeyPath? = nil, countEquals count: Int) {
```

Comment:
For these `count` variants, is it possible to reuse your other extensions and add "@count" to the provided keyPath, somehow? Might be clunky.